### PR TITLE
Rename backendStoreDriver to dataEngin in instance-manager and associated components

### PR DESCRIPTION
**Pull request by derekbit**
https://github.com/longhorn/longhorn/issues/7095
https://github.com/longhorn/longhorn/issues/7480

deps on https://github.com/longhorn/backupstore/pull/153 and https://github.com/longhorn/longhorn-instance-manager/pull/348

### CODE REVIEW

File: engineapi/instance_manager.go [hunk 1]
```diff
@@ -240,10 +240,22 @@ func parseProcess(p *imapi.Process) *longhorn.InstanceProcess {
 			Conditions: p.ProcessStatus.Conditions,
 			PortStart:  p.ProcessStatus.PortStart,
 			PortEnd:    p.ProcessStatus.PortEnd,
+
+			// These fields are not used, maybe we can deprecate them later.
```
Comment (innobead) [inline engineapi/instance_manager.go:244]: Sounds good. 

// FIXME These fields are not used, maybe we can deprecate them later.

**Review by innobead — APPROVED** [review]
Merge this PR after the dep gets updated.

File: engineapi/instance_manager.go (hunk 1 above)
Comment (derekbit) [reply engineapi/instance_manager.go]: Updated. Thank you!

**Review by innobead — APPROVED** [review]

File: engineapi/instance_manager.go [hunk 2]
```diff
@@ -240,10 +240,22 @@ func parseProcess(p *imapi.Process) *longhorn.InstanceProcess {
...
 		},
 	}
 }
 
+func getDataEngineFromInstanceProcess(p *imapi.Instance) longhorn.DataEngineType {
+	if p.DataEngine != "" {
+		return longhorn.DataEngineType(p.DataEngine)
+	}
+
+	return longhorn.DataEngineType(p.BackendStoreDriver)
```
Comment (innobead) [inline engineapi/instance_manager.go:256]: Caught by the PR check.

Comment (derekbit) [reply engineapi/instance_manager.go]: Fixed. Thank you.

File: engineapi/instance_manager.go [hunk 3]
```diff
@@ -240,10 +240,22 @@ func parseProcess(p *imapi.Process) *longhorn.InstanceProcess {
...
 		},
 	}
 }
 
+func getDataEngineFromInstanceProcess(p *imapi.Instance) longhorn.DataEngineType {
+	if p.DataEngine != "" {
+		return longhorn.DataEngineType(p.DataEngine)
+	}
+
+	return longhorn.DataEngineType(p.DataEngine)
```
Comment (c3y1huang) [inline engineapi/instance_manager.go:256]: This returns the same value as the condition above?

Comment (innobead) [reply engineapi/instance_manager.go]: Good :eye:

Comment (derekbit) [reply engineapi/instance_manager.go]: The original one is correct. That is falling back to backendStoreDriver.
static checker does not like reading data from deprecated field...

Comment (derekbit) [reply engineapi/instance_manager.go]: Any solution to the golinter's deprecation check?
@innobead @c3y1huang

Comment (c3y1huang) [reply engineapi/instance_manager.go]: I think we could ignore `SA1019`? 
https://github.com/longhorn/longhorn-manager/pull/2400/files#diff-4c0e1e708c59d9a06da4c01a46c2d30739fce6b57e17eff33b2930da8d294692R25

Comment (innobead) [reply engineapi/instance_manager.go]: Can we ignore a specific line? I feel it should be feasible.

Comment (innobead) [reply engineapi/instance_manager.go]: https://golangci-lint.run/usage/false-positives/#nolint-directive

Comment (derekbit) [reply engineapi/instance_manager.go]: Not sure why `// lint:ignore SA1019` doesn't work. Let me try `nolint`

Comment (derekbit) [reply engineapi/instance_manager.go]: `nolint: all` works

**Review by innobead — APPROVED** [review]

**Pull request merged**